base/common/python/pki/info.py

```diff
@@ -119,11 +119,19 @@ def __init__(self, parent):
         else:
             self.pki_client = parent
             self.connection = self.pki_client.connection
+        self.api_path = None
+        self.api_version = None
 
     @pki.handle_exceptions()
     def get_info(self):
         """ Return an Info object form a PKI server """
 
+        api_versions = {
+            'rest': 'v1',
+            'v1': 'v1',
+            'v2': 'v2',
+        }
+
```

Comment:
Not sure this can be used this way. The PKI version check should be done before setting the api_verision to use so it is better to let the client to define what to use.

Reply:
If existing IPA users can't be required to upgrade, then you might want to create `IPAClient` object with `api_version='v1'` explicitly when connecting to those legacy servers. But as mentioned above, APIv1 will eventually be dropped at some point in the future.

Right now we haven't identified anything that needs specific version check other than `>= 11.6.0`, but since APIv2 will continuously be developed, there might be more in the future. When a new feature is added it doesn't mean it's immediately ready to be used widely (e.g. a tech preview), so we can use the version check to see whether the new feature is mature enough to use.